crates/bevy_winit/Cargo.toml

```diff
@@ -18,6 +19,7 @@ bevy_app = { path = "../bevy_app", version = "0.9.0-dev" }
 bevy_ecs = { path = "../bevy_ecs", version = "0.9.0-dev" }
 bevy_input = { path = "../bevy_input", version = "0.9.0-dev" }
 bevy_math = { path = "../bevy_math", version = "0.9.0-dev" }
+bevy_render = { path = "../bevy_render", version = "0.9.0-dev" }
```

Comment:
This is probably undesirable because bevy_winit is technically "lower level" than bevy_render is .Could we move this to another crate?

Reply:
Not sure. We need a place to call setup rendering once and only once after all the plugins call their build function. I was looking, but haven't found it yet. I guess an option might be to store some state in the update_rendering function and `setup_rendering` on the first call to it. Does that sound better?

Reply:
actually `update_rendering` doesn't have a reference to app to remove the sub app. Need to keep looking for a place to do this.

Reply:
looking at the dependency graph bevy_winit seems to be at the same level as bevy_render. bevy_window is the one lower down.

left is main and right is this pr
![image](https://user-images.githubusercontent.com/2180432/201257159-d07a1382-be85-4cb5-af90-4e93bac3d3e2.png)

Reply:
My plan for this now is to add another callback to subapps that will run once before calling the runner or extract.

Reply:
Added a setup callback to plugins that is called during App::run before the runner is called..